typescript/sdk/src/gas/token-prices.ts

```diff
@@ -153,4 +155,32 @@ export class CoinGeckoTokenPriceGetter implements TokenPriceGetter {
     // Update the cache with the newly fetched prices
     chains.map((chain, i) => this.cache.put(chain, prices[i]));
   }
+
+  public async getTokenPriceByIds(
```

Comment:
The try catch behavior could be pulled up into calling code instead of being here. The problem with this is that it can mask the reason for why the query failed. Sometimes we are rate limited and the cache is not updated as we don't have a price. We then have a cache miss which raises the exception that is bubbled up